### PR TITLE
feat: Extract google-cloud-translate-v2 from google-cloud-translate

### PR DESCRIPTION
**Pull request by dazuma**
Translate V2 is a snowflake in that there is no gRPC endpoint and no public protos. The client is completely handwritten directly atop an HTTP client.

This PR copies the Translate V2 handwritten client from `google-cloud-translate` into a separate gem `google-cloud-translate-v2`. The code has been copied verbatim, with an unaltered interface. However, the following caveats apply:

* Previously, the code could assume that `Google::Cloud.config.translate` was available. However, that configuration belongs to `google-cloud-translate`, and may not be present if `google-cloud-translate-v2` is installed in isolation. So the configuration interface (in `Google::Cloud::Translate::V2.new`) has been adjusted to handle the case where `Google::Cloud.config.translate` is not present. In that case, we simply use the defaults that the config would have picked up from environment variables (such as `TRANSLATE_KEY` and `TRANSLATE_PROJECT`).
* The entrypoint `require "google/cloud/translate"` belongs to the `google-cloud-translate` gem, and again might not be present if `google-cloud-translate-v2` is installed in isolation. Adjusted all documentation, examples, and tests to `require "google/cloud/translate/v2"` instead. Similarly, the `Google::Cloud::Translate.new` method belongs to `google-cloud-translate`, so all docs, examples, and tests have been adjusted to use `Google::Cloud::Translate::V2.new`.
* Following the convention of our microgenerator-produced clients `require "google-cloud-translate-v2"` is a noop. Users should instead `require "google/cloud/translate/v2"`.
* A few of the auxiliary files (such as `.gitignore` and `README.md`) have been updated to the standard formats produced by the microgenerator.
* As this client is completely handwritten, there is no `synth.py` script.
* All tests, doctests, and acceptance tests still pass.
* I updated tests to conform to the updated minitest spec format `_(expr).must_be_something`.
* I kept the release history from `google-cloud-translate` in the changelog, with a header delimiting when the switch was made.

The plan is for `google-cloud-translate-v2` to be treated like any of our microgenerator-produced clients. (Its sibling, `google-cloud-translate-v3`, is microgenerator-produced.) The `google-cloud-translate` gem will then be turned into a wrapper.

### CODE REVIEW

File: google-cloud-translate-v2/Rakefile [hunk 1]
```diff
@@ -0,0 +1,187 @@
+# frozen_string_literal: true
```
Comment (quartzmo) [inline google-cloud-translate-v2/Rakefile:1]: Do you want `frozen_string_literal: true` in these files? It appears in some of them but not most.

File: google-cloud-translate-v2/README.md [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+translation.to #=> "la"
+translation.text #=> "Salve mundi!"
+```
+
+View the [Client Library Documentation](https://googleapis.dev/ruby/google-cloud-translate-v2/latest)
+for class and method documentation.
+
+## Enabling Logging
+
+To enable logging for this library, set the logger for the underlying [gRPC](https://github.com/grpc/grpc/tree/master/src/ruby) library. The logger that you set may be a Ruby stdlib [`Logger`](https://ruby-doc.org/stdlib-2.5.0/libdoc/logger/rdoc/Logger.html) as shown below, or a [`Google::Cloud::Logging::Logger`](https://googleapis.dev/ruby/google-cloud-logging/latest) that will write logs to [Stackdriver Logging](https://cloud.google.com/logging/). See [grpc/logconfig.rb](https://github.com/grpc/grpc/blob/master/src/ruby/lib/grpc/logconfig.rb) and the gRPC [spec_helper.rb](https://github.com/grpc/grpc/blob/master/src/ruby/spec/spec_helper.rb) for additional information.
```
Comment (quartzmo) [inline google-cloud-translate-v2/README.md:45]: gRPC logging won't work in this lib, will it?

File: google-cloud-translate-v2/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
+# Release History
+
+## Previous releases of the google-cloud-translate gem
```
Comment (quartzmo) [inline google-cloud-translate-v2/CHANGELOG.md:3]: Hmmm, I think I would prefer to leave out this history, and just start with a blank slate. But that's just opinion.

File: google-cloud-translate-v2/AUTHENTICATION.md [hunk 4]
```diff
@@ -0,0 +1,181 @@
+# Authentication
+
+In general, the google-cloud-translate library uses [Service
```
Comment (quartzmo) [inline google-cloud-translate-v2/AUTHENTICATION.md:3]: Replace with google-cloud-translate-v2 here?

File: google-cloud-translate-v2/AUTHENTICATION.md [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+environments, the Service Account credentials can be specified by providing the
+path to the [JSON
+keyfile](https://cloud.google.com/iam/docs/managing-service-account-keys) for
+the account (or the JSON itself) in environment variables. Additionally, Cloud
+SDK credentials can also be discovered automatically, but this is only
+recommended during development.
+
+## Project and Credential Lookup
+
+The google-cloud-translate library aims to make authentication as simple as
```
Comment (quartzmo) [inline google-cloud-translate-v2/AUTHENTICATION.md:18]: Replace with google-cloud-translate-v2 here?

File: google-cloud-translate-v2/README.md [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+1. [Select or create a Cloud Platform project.](https://console.cloud.google.com/project)
+1. [Enable billing for your project.](https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project)
+1. {file:AUTHENTICATION.md Set up authentication.}
+
+## Quick Start
+
+```ruby
+require "google/cloud/translate/v2"
+
+client = ::Google::Cloud::Translate::V2.new
```
Comment (quartzmo) [inline google-cloud-translate-v2/README.md:28]: We don't usually use `::Google` scoping in examples. Is it needed here?

File: google-cloud-translate-v2/AUTHENTICATION.md [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+
+```ruby
+require "google/cloud/translate"
+
+Google::Cloud::Translate.configure do |config|
+  config.project_id  = "my-project-id"
+  config.credentials = "path/to/keyfile.json"
+end
+
+translate = Google::Cloud::Translate.new
```
Comment (quartzmo) [inline google-cloud-translate-v2/AUTHENTICATION.md:113]: Should this be `Google::Cloud::Translate::V2.new`?

File: google-cloud-translate-v2/AUTHENTICATION.md [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+The **Project ID** and **Credentials JSON** can be configured instead of placing them in environment variables or providing them as arguments.
+
+```ruby
+require "google/cloud/translate"
+
+Google::Cloud::Translate.configure do |config|
+  config.project_id  = "my-project-id"
+  config.credentials = "path/to/keyfile.json"
+end
```
Comment (quartzmo) [inline google-cloud-translate-v2/AUTHENTICATION.md:111]: Shouldn't this configuration be passed directly to `Google::Cloud::Translate::V2.new`?

File: google-cloud-translate-v2/Rakefile (hunk 1 above)
Comment (dazuma) [reply google-cloud-translate-v2/Rakefile]: Removed.

File: google-cloud-translate-v2/README.md (hunk 2 above)
Comment (dazuma) [reply google-cloud-translate-v2/README.md]: No it won't! Removed.

File: google-cloud-translate-v2/CHANGELOG.md (hunk 3 above)
Comment (dazuma) [reply google-cloud-translate-v2/CHANGELOG.md]: Ehhhh... I don't feel strongly. I'll remove it.

File: google-cloud-translate-v2/AUTHENTICATION.md (hunk 4 above)
Comment (dazuma) [reply google-cloud-translate-v2/AUTHENTICATION.md]: fixed.

File: google-cloud-translate-v2/AUTHENTICATION.md (hunk 5 above)
Comment (dazuma) [reply google-cloud-translate-v2/AUTHENTICATION.md]: fixed.

File: google-cloud-translate-v2/AUTHENTICATION.md (hunk 7 above)
Comment (dazuma) [reply google-cloud-translate-v2/AUTHENTICATION.md]: Removed this section.

File: google-cloud-translate-v2/AUTHENTICATION.md (hunk 8 above)
Comment (dazuma) [reply google-cloud-translate-v2/AUTHENTICATION.md]: Removed this section.

File: google-cloud-translate-v2/README.md (hunk 6 above)
Comment (dazuma) [reply google-cloud-translate-v2/README.md]: Updated.

**Comment by dazuma** [comment]
Ugh, I did a sloppy job of that one. Thanks!

**Pull request merged**